docs/build-clio.md

```diff
@@ -88,6 +88,31 @@ If successful, `conan install` will find the required packages and `cmake` will
 > [!TIP]
 > To generate a Code Coverage report, include `-o coverage=True` in the `conan install` command above, along with `-o tests=True` to enable tests. After running the `cmake` commands, execute `make clio_tests-ccov`. The coverage report will be found at `clio_tests-llvm-cov/index.html`.
 
+### Generating API docs for Clio
+
+The API documentation for Clio is generated by [Doxygen](https://www.doxygen.nl/index.html). If you want to generate the API documentation when building Clio, make sure to install Doxygen on your system.
+
+To generate the API docs:
+
+1. First, include `-o docs=True` in the conan install command. For example:
+
+    ```sh
+    conan install .. --output-folder . --build missing --settings build_type=Release -o tests=True -o lint=False -o docs=True
+    ```
+
+2. Once that has completed successfully, run the `cmake` command and add the `--target docs` option:
+
+   ```sh
+   cmake -DCMAKE_TOOLCHAIN_FILE:FILEPATH=build/generators/conan_toolchain.cmake -DCMAKE_BUILD_TYPE=Release ..
+   cmake --build . --parallel 8 --target docs
+   ```
+
+3. In the `build` directory, you should see a folder called `html`. Go to the folder to view the generated files.
+
+    Open the `index.html` file in your browser to see the documentation pages.
+
+    ![API index page](./img/doxygen-docs-output.png "Title")
```

Comment:
I think the doc update is good. I'm also ok with adding the screenshot - doxygen theme will be changed later and we can update the screenshot then.
What's not great is to upload binary files like images to git. Normally people use something like `git-lfs` for that. So i think we should merge something [like this](https://github.com/XRPLF/clio/pull/1210) first and then rebase your changes on top of develop once ready.

Reply:
The problem may be that we then also need to document `git-lfs` as a dependancy for development of Clio. I think CI is passing still so perhaps `git-lfs` is not strictly needed unless you touch the binary files.

Reply:
I think we can update the screenshot now and also maybe mention https://xrplf.github.io/clio somewhere - that's where the docs are always available for latest `develop` branch.